signature-aggregator/aggregator/aggregator.go

```diff
@@ -459,8 +462,9 @@ func (s *SignatureAggregator) isValidSignatureResponse(
 		)
 		return blsSignatureBuf{}, false
 	}
-
-	return signature, true
+	blsSig := blsSignatureBuf{}
+	copy(blsSig[:], signature[:])
```

Comment:
Should we have a check higher up that the length of `signature` is the expected length?

Reply:
Possibly, we do call `bls.SignatureFromBytes` and `bls.Verify` on it's result so presumably those wouldn't work if it wasn't correct lenght but an explicit check wouldn't hurt

Reply:
Addressed in 811c97a